gorm/transaction.go

```diff
@@ -173,10 +175,18 @@ func (t *Transaction) Commit(ctx context.Context) error {
 // Client is responsible to call `txn.Begin()` to open transaction.
 // If call of grpc.UnaryHandler returns with an error the transaction
 // is aborted, otherwise committed.
-func UnaryServerInterceptor(db *gorm.DB) grpc.UnaryServerInterceptor {
+func UnaryServerInterceptor(db *gorm.DB, opts ...Option) grpc.UnaryServerInterceptor {
```

Comment:
Try adding a new func that takes a Transaction. You would then have the ability to register a hook. 
```
 txn := NewTransaction(db)
 txn.AddAfterCommitHook(...myhook...)
  
 UnaryServerInterceptorTxn(txn) grpc.UnaryServerInterceptor {
 ...
 }
```